### ISSUE TITLE
-M option reports the wrong number of reads processed

### ISSUE THREAD

**Issue by dominik-handler** [BenLangmead/bowtie#125]
Hi,

I realized a problem in the reporting of mapping statistics if option -M is used. I have a read-set that only contains multi-mapping reads. When I run with -m 1 I correctly get all reads suppressed and the correct number of reads reported. If I use option -M 1 it seems that instead of the read-number the total mapping number is reported as read-count:

```
bowtie -p 12 -f -S -v 2 --best --strata -m 1 index - > x.sam
# reads processed: 7078925
# reads with at least one reported alignment: 0 (0.00%)
# reads that failed to align: 533923 (7.54%)
# reads with alignments suppressed due to -m: 6545002 (92.46%)
No alignments

bowtie -p 12 -f -S -v 2 --best --strata -M 1 index - > x.sam
# reads processed: 13623927
# reads with at least one reported alignment: 6545002 (48.04%)
# reads that failed to align: 533923 (3.92%)
# reads with alignments sampled due to -M: 6545002 (48.04%)
Reported 6545002 alignments
```

Otherwise the output looks fine to me. 

Thank you, 
Dominik

**Comment by ch4rr0** [comment]
Hello @dominik-handler,

I committed an issue that I think fixes the issue. Here's the output from my test runs:

``` text
./bowtie-align-s -x indexes/e_coli reads/e_coli_1000.fq -m1 --best -S
# reads processed: 1000
# reads with at least one alignment: 699 (69.90%)
# reads that failed to align: 301 (30.10%)
# reads with alignments suppressed due to -m: 22 (2.20%)
Reported 677 alignments

./bowtie-align-s -x indexes/e_coli reads/e_coli_1000.fq -M1 --best -S
# reads processed: 1000
# reads with at least one alignment: 699 (69.90%)
# reads that failed to align: 301 (30.10%)
# reads with alignments sampled due to -M: 22 (2.20%)
Reported 699 alignments
```

Please let me know your thoughts on this.

**Comment by ch4rr0** [comment]
This change is now available in our latest release, v1.3.1. Feel free to reopen the issue if you feel like it has not been properly addressed.
